app/reducers/usersActivity.js

```diff
@@ -0,0 +1,52 @@
+import { USERS_ACTIVITY } from '../actions/actionsTypes';
+
+/**
+ * The state could be
+ * {
+ *	roomId?: {
+ *		username: action
+ *	},
+ *	tmid?: {
+ *		username: action
+ *	}
+ * }
+ */
+
+const initialState = {};
+
+export default function usersActivity(state = initialState, action) {
+	const { type, roomId, username, activity } = action;
+
+	switch (type) {
+		case USERS_ACTIVITY.ADD:
+			if (state[roomId]) {
+				const obj = state[roomId];
+				delete obj[username];
```

Comment:
Reducers should not mutate the state https://redux.js.org/usage/structuring-reducers/prerequisite-concepts

This line of code was making events forEach to be unpredictable https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3243/files#diff-67b9b270a5ff6f8faa4f97992f40a09631edb29f76b5201cb83d491566bc006aR131